internal/pkg/controllers/profilerecorder/profilerecorder.go

```diff
@@ -0,0 +1,249 @@
+		return errors.Wrap(err, errGetNode)
+	}
+
+	nodeAddress := ""
+	for _, addr := range node.Status.Addresses {
+		if addr.Type == corev1.NodeInternalIP {
+			l.Info("Setting up profile recorder", "Node", addr.Address)
+			nodeAddress = addr.Address
+			break
+		}
+	}
+
+	if nodeAddress == "" {
+		return errors.New("Unable to get node's internal Address")
+	}
+
+	reconiler := RecorderReconciler{
+		client:      mgr.GetClient(),
+		log:         l,
+		nodeAddress: nodeAddress,
+		record:      event.NewAPIRecorder(mgr.GetEventRecorderFor(name)),
+		podsToWatch: map[string]string{},
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithEventFilter(predicate.And(
+			resource.NewPredicates(reconiler.isPodWithTraceAnnotation),
+			resource.NewPredicates(reconiler.isPodOnLocalNode),
+		)).
+		For(&corev1.Pod{}).
+		Complete(&reconiler)
+}
+
+func (r *RecorderReconciler) isPodOnLocalNode(obj runtime.Object) bool {
+	p, ok := obj.(*corev1.Pod)
+
+	if !ok {
+		return false
+	}
+
+	return p.Status.HostIP == r.nodeAddress
+}
+
+func (r *RecorderReconciler) isPodWithTraceAnnotation(obj runtime.Object) bool {
+	p, ok := obj.(*corev1.Pod)
+
+	if !ok {
+		return false
+	}
+
+	return p.Annotations[config.SeccompProfileRecordAnnotationKey] != ""
+}
+
+func (r *RecorderReconciler) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	logger := r.log.WithValues("recorder", req.Name, "namespace", req.Namespace)
+
+	ctx, cancel := context.WithTimeout(context.Background(), reconcileTimeout)
+	defer cancel()
+
+	pod := &corev1.Pod{}
+	if err := r.client.Get(ctx, req.NamespacedName, pod); err != nil {
+		if kerrors.IsNotFound(err) {
+			if err := r.collectProfile(ctx, req.NamespacedName); err != nil {
+				return reconcile.Result{}, nil
+			}
+		} else {
+			// Returning an error means we will be requeued implicitly.
+			logger.Error(err, "Error reading pod")
+			return reconcile.Result{}, errors.Wrap(err, errGetPod)
+		}
+	}
+
+	if pod.Status.Phase == corev1.PodRunning || pod.Status.Phase == corev1.PodSucceeded {
+		if r.podsToWatch[req.NamespacedName.String()] != "" {
+			// We're tracking this pod already
+			return reconcile.Result{}, nil
+		}
+		outputFile, err := parseAnnotation(pod.Annotations[config.SeccompProfileRecordAnnotationKey])
+		if err != nil {
+			logger.Info("Unable to parse annoation. Ingoring", "error", err)
+			return reconcile.Result{}, nil
+		}
+		if !strings.HasPrefix(outputFile, config.SeccompProfileRecordOutputPath) {
+			logger.Info("Ingoring Profile outside standard output path")
+		}
+		r.podsToWatch[req.NamespacedName.String()] = outputFile
+		r.record.Event(pod, event.Normal(reasonProfileRecording, "Recording seccomp profile"))
+	}
+	if pod.Status.Phase == corev1.PodSucceeded {
+		if err := r.collectProfile(ctx, req.NamespacedName); err != nil {
+			return reconcile.Result{}, nil
+		}
+	}
+
+	for name, profile := range r.podsToWatch {
+		logger.Info("RecorderReconciler", "Pod", name, "Profile", profile)
```

Comment:
Is there a specific reason to just `log` within the loop. If there is a need, probably a comment would help.

Reply:
No this was a debugging leftover. I removed that loop completely.